tools/pipelines/templates/download-pipeline-artifact.yml

```diff
@@ -0,0 +1,29 @@
+# Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+# Licensed under the MIT License.
+
+# download-pipeline-artifact pipeline
+
+parameters:
+- name: pipelineDefinition
+  type: string
+
+- name: branchName
+  type: string
```

Comment:
I'll get ahead of @Josmithr and ask for a bit of documentation for the parameters here 😄. Nit: I'd also suggest `pipelineName` instead of `pipelineDefinition`.

Also, I think it works to add a default for `branchName` here so we don't have to pass the same value each time we "call" the template:

```yaml
- name: branchName
  type: string
  default: ${{ variables['Build.SourceBranch'] }}
```

Use of `variables.<something>` with compile-time syntax (`${{ }}`) has caveats but _I think_ in this case, for this particular variable, it's fine.